packages/service-client/odsp-client/src/odspClient.ts

```diff
@@ -0,0 +1,205 @@
+ * Licensed under the MIT License.
+ */
+import { IDocumentServiceFactory } from "@fluidframework/driver-definitions";
+import {
+	OdspDocumentServiceFactory,
+	OdspDriverUrlResolver,
+	createOdspCreateContainerRequest,
+} from "@fluidframework/odsp-driver";
+import {
+	type ContainerSchema,
+	DOProviderContainerRuntimeFactory,
+	IFluidContainer,
+	FluidContainer,
+	IRootDataObject,
+} from "@fluidframework/fluid-static";
+import {
+	AttachState,
+	IContainer,
+	IFluidModuleWithDetails,
+} from "@fluidframework/container-definitions";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { v4 as uuid } from "uuid";
+import {
+	IOdspResolvedUrl,
+	OdspResourceTokenFetchOptions,
+} from "@fluidframework/odsp-driver-definitions";
+import { ITokenResponse } from "@fluidframework/azure-client";
+// eslint-disable-next-line import/no-deprecated
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { IRequest } from "@fluidframework/core-interfaces";
+import {
+	OdspClientProps,
+	OdspConnectionConfig,
+	OdspContainerServices,
+	OdspServiceAttributes,
+} from "./interfaces";
+import { OdspAudience } from "./odspAudience";
+
+/**
+ * OdspClient provides the ability to have a Fluid object backed by the ODSP service within the context of Microsoft 365 (M365) tenants.
+ *
+ * @alpha @sealed
+ */
+export class OdspClient {
+	private readonly documentServiceFactory: IDocumentServiceFactory;
+	private readonly urlResolver: OdspDriverUrlResolver;
+
+	public constructor(private readonly properties: OdspClientProps) {
+		const getSharePointToken = async (options: OdspResourceTokenFetchOptions) => {
+			const tokenResponse: ITokenResponse =
+				await this.properties.connection.tokenProvider.fetchStorageToken(
+					options.siteUrl,
+					"",
+				);
+			return {
+				token: tokenResponse.jwt,
+			};
+		};
+
+		const getPushServiceToken = async (options: OdspResourceTokenFetchOptions) => {
+			const tokenResponse: ITokenResponse =
+				await this.properties.connection.tokenProvider.fetchOrdererToken(options.siteUrl);
+			return {
+				token: tokenResponse.jwt,
+			};
+		};
+		this.documentServiceFactory = new OdspDocumentServiceFactory(
+			getSharePointToken,
+			getPushServiceToken,
+		);
+
+		this.urlResolver = new OdspDriverUrlResolver();
+	}
+
+	public async createContainer(containerSchema: ContainerSchema): Promise<{
+		container: IFluidContainer;
+		services: OdspContainerServices;
+	}> {
+		const loader = this.createLoader(containerSchema);
+
+		const container = await loader.createDetachedContainer({
+			package: "no-dynamic-package",
+			config: {},
+		});
+
+		const fluidContainer = await this.createFluidContainer(
+			container,
+			this.properties.connection,
+		);
+
+		const services = await this.getContainerServices(container);
+
+		return { container: fluidContainer, services };
+	}
+
+	public async getContainer(
+		sharingUrl: string,
```

Comment:
I've left comments in document in PR about it: 
1. I'm not sure it has to be sharing link (and types of sharing links we support). It has perf impact.
2. I do not understand how it all works if URL points to different drive ID or different siteURI / path that are specified when OdspClient is created. Do we ignore them? What driveID is returned on different APIs - one that was supplied to OdspClient, or one that identifies a file?
3. I think we are confusing users about exposing too many concepts, like URI, containerID, itemID. I think we should expose one, and provide static functions to get all the other data.
4. I assume we will not allow users to provide their own URI resolvers, and leave them to do conversion of their app API -> our ID/path outside of these interactions (but it would be worth confirming and describing it somewhere)

Reply:
Regarding (2): 
`I do not understand how it all works if URL points to different drive ID or different siteURI / path that are specified when OdspClient is created. Do we ignore them?` - I'll look into this more. 

`What driveID is returned on different APIs - one that was supplied to OdspClient, or one that identifies a file?` - Does this mean there will be two separate drive ids? Why would there be a different driveId to identify a file?
You can only use one driveId for any given Fluid Container. So, if you need to use more, you can create additional odsp-clients. But the Drive ID will be different for each tenant.

Reply:
@ChumpChief @vladsud  I checked the way azure client is doing this. It seems like that id is not enough there too. The azure endpoint is taken in constructor and then using this id we make some modifications to that endpoint and append this id to reach to the request url which is then used to load the container. I am not sure I like that a lot because that is kind of functionality provided by resolver and its utility apis. Similarly, ODSP also has endpoint encoding all this info but imo it is not a sound way to arrive at the request url. 

I don't see any harm in taking the url when app can make use of the utility api to construct it and just pass it to this function. And it does not necessarily be a sharing link, getAbsoluteUrl in OdspDriverUrlResoover does not give back a asharing link, but it is still enough to load the container as the absolute url is resolvable by the same resolver.